src/urllib3/poolmanager.py

```diff
@@ -537,7 +537,7 @@ class ProxyManager(PoolManager):
 
     def __init__(
         self,
-        proxy_url: str,
+        proxy_url: Union[HTTPConnectionPool, str],
```

Comment:
It's funky that an argument called `proxy_url` can be an `HTTPConnectionPool` but I'm sure there's a story there :grin:

Reply:
Hmm yeah maybe we shouldn't advertise this.

Reply:
@hramezani Can we remove the type hint for `HTTPConnectionPool` since it's a fallback to cover old behavior? Instead we can add `type: ignore` to the test suite.